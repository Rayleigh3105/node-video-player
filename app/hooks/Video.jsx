/* eslint-disable jsx-a11y/media-has-caption */
import React from 'react';
import PropTypes from 'prop-types';

const Video = ({ src }) => {
  const extension = src.split('.').pop();
  return (
    <div className={`row video-container ${src !== '' ? 'visible' : ''}`}>
      <div className="col-md-12">
        <video autoPlay controls width="100%" key={src}>
          <source src={`/videos/${src}`} type={`video/${extension}`} />
        </video>
      </div>
    </div>
  );
};

Video.propTypes = {
  src: PropTypes.string.isRequired,
};

export default Video;
