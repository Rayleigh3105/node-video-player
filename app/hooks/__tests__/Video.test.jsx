import { create, act } from 'react-test-renderer';
import React from 'react';
import Video from '../Video';

it('should render correctly', () => {
  let root;
  act(() => {
    root = create(<Video src="file.mp4" />);
  });

  expect(root.toJSON()).toMatchSnapshot();
});
