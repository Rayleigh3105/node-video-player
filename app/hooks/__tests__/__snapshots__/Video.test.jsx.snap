// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="row video-container visible"
>
  <div
    className="col-md-12"
  >
    <video
      autoPlay={true}
      controls={true}
      width="100%"
    >
      <source
        src="/videos/file.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;
